packages/components/src/ui/ariakit-tooltip/index.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { Tooltip, TooltipAnchor, useTooltipState } from 'ariakit/tooltip';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement, isValidElement } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import { TOOLTIP_DELAY } from '../../tooltip/';
+import type { ToolTipProps } from './types';
+import Shortcut from '../../shortcut';
+import { positionToPlacement as __experimentalPopoverLegacyPositionToPlacement } from '../../popover/utils';
+import * as styles from './styles';
+import { contextConnectWithoutRef } from '../context/context-connect';
+import { useCx } from '../../utils/hooks/use-cx';
+
+function AriaToolTip( props: ToolTipProps ) {
+	const {
+		children,
+		delay = TOOLTIP_DELAY,
+		placement,
+		position,
+		shortcut,
+		text,
+	} = props;
+
+	const baseId = useInstanceId( ToolTip, 'tooltip' );
+	const describedById = text || shortcut ? baseId : undefined;
+
+	const DEFAULT_PLACEMENT = 'bottom';
+
+	// Compute tooltip's placement:
+	// - give priority to `placement` prop, if defined
+	// - otherwise, compute it from the legacy `position` prop (if defined)
+	// - finally, fallback to the DEFAULT_PLACEMENT.
+	let computedPlacement;
+	if ( placement !== undefined ) {
+		computedPlacement = placement;
+	} else if ( position !== undefined ) {
+		computedPlacement =
+			// @ts-expect-error
```

Comment:
This error is happening because the legacy to position function only accepts a range of values representing only valid Popover `position`s.

The way we can fix this is to set the correct types for the `position` and `placement` props:
- for the `placement` prop, we can borrow the types from the ariakit tooltip
- for the `position` prop, we can borrow types from `Popover`


```diff
diff --git a/packages/components/src/ui/ariakit-tooltip/index.tsx b/packages/components/src/ui/ariakit-tooltip/index.tsx
index 4ff8859de0..178bd76aab 100644
--- a/packages/components/src/ui/ariakit-tooltip/index.tsx
+++ b/packages/components/src/ui/ariakit-tooltip/index.tsx
@@ -45,7 +45,6 @@ function AriaToolTip( props: ToolTipProps ) {
 		computedPlacement = placement;
 	} else if ( position !== undefined ) {
 		computedPlacement =
-			// @ts-expect-error
 			__experimentalPopoverLegacyPositionToPlacement( position );
 	}
 	computedPlacement = computedPlacement || DEFAULT_PLACEMENT;
diff --git a/packages/components/src/ui/ariakit-tooltip/types.ts b/packages/components/src/ui/ariakit-tooltip/types.ts
index d7684710fe..d4bde191d3 100644
--- a/packages/components/src/ui/ariakit-tooltip/types.ts
+++ b/packages/components/src/ui/ariakit-tooltip/types.ts
@@ -3,6 +3,11 @@
  */
 import type { TooltipStateProps } from 'ariakit/tooltip';
 
+/**
+ * Internal dependencies
+ */
+import type { PopoverProps } from '../../popover/types';
+
 export type ToolTipProps = Pick< TooltipStateProps, 'placement' > & {
 	/**
 	 * The anchor for the tooltip. Accepts only one child element.
@@ -19,7 +24,7 @@ export type ToolTipProps = Pick< TooltipStateProps, 'placement' > & {
 	 *
 	 * @default
 	 */
-	placement?: any; //Placement | undefinded;
+	placement?: TooltipStateProps[ 'placement' ];
 	/**
 	 * _Note: this prop is deprecated. Please use the `placement` prop instead._
 	 *
@@ -27,7 +32,7 @@ export type ToolTipProps = Pick< TooltipStateProps, 'placement' > & {
 	 *
 	 * @default top center
 	 */
-	position?: string;
+	position?: PopoverProps[ 'position' ];
 	/**
 	 * Option for adding accessible keyboard shortcuts.
 	 */

```

But we'll still incur in a TS error, because the return type of `__experimentalPopoverLegacyPositionToPlacement` currently also include a special placement that we introduced in our `Popover` (called `overlay`). This placement won't work for the ariakit popover (ie. _vanilla_ floating-ui, without any customization).

The best way to fix this is to actually realise that the `positionToPlacement` function currently doesn't return `overlay` in any scenario — we can then update its types, and fix the TS error in tooltip!


```diff
diff --git a/packages/components/src/popover/utils.ts b/packages/components/src/popover/utils.ts
index 022e7796fe..1e03c9ed19 100644
--- a/packages/components/src/popover/utils.ts
+++ b/packages/components/src/popover/utils.ts
@@ -16,7 +16,7 @@ import type {
 
 const POSITION_TO_PLACEMENT: Record<
 	NonNullable< PopoverProps[ 'position' ] >,
-	NonNullable< PopoverProps[ 'placement' ] >
+	NonNullable< Exclude< PopoverProps[ 'placement' ], 'overlay' > >
 > = {
 	bottom: 'bottom',
 	top: 'top',
@@ -79,8 +79,7 @@ const POSITION_TO_PLACEMENT: Record<
  */
 export const positionToPlacement = (
 	position: NonNullable< PopoverProps[ 'position' ] >
-): NonNullable< PopoverProps[ 'placement' ] > =>
-	POSITION_TO_PLACEMENT[ position ] ?? 'bottom';
+) => POSITION_TO_PLACEMENT[ position ] ?? 'bottom';
 
 /**
  * @typedef AnimationOrigin

```

We can actually update the types of `POSITION_TO_PLACEMENT` and then remove the explicit return type from the `positionToPlacement` function — typescript will be able to infer them correctly.